src/Table/components/TableHeader.tsx

```diff
@@ -33,6 +35,7 @@ export const TableHeader = <T extends object>({
           minWidth={rowActions?.minWidth}
           headerColor={headerColor}
           columnPadding={columnPadding}
+          hasKeyline={hasKeyline}
```

Comment:
Can you please add this prop as a [transient prop](https://styled-components.com/docs/api#transient-props)? It's basically just adding a `$` sign before the property name. We have [this PR](https://github.com/marshmallow-insurance/smores-react/pull/3555) open where `styled-components` version is being bumped and every prop is transformed to transient prop.